src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -294,6 +297,39 @@ public FileResource(FileService fileService, ResourceLoaderService resourceLoade
         return buildFileResponse(Paths.get(FilePathService.getLectureAttachmentFilePath(), String.valueOf(optionalLecture.get().getId())).toString(), filename);
     }
 
+    /**
+     * GET /files/attachments/lecture/{lectureId}/merge-pdf : Get the lecture units PDF attachments merged
+     *
+     * @param lectureId ID of the lecture, the lecture units belongs to
+     * @param temporaryAccessToken The access token is required to authenticate the user that accesses it
+     * @return The merged pdf file, 403 if the logged in user is not allowed to access it, or 404 if the files to be merged doesn't exist
+     */
+    @GetMapping("files/attachments/lecture/{lectureId}/merge-pdf")
+    @PreAuthorize("permitAll()")
+    public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long lectureId, @RequestParam("access_token") String temporaryAccessToken) {
+        log.debug("REST request to get merged pdf files for a lecture with id : {}", lectureId);
+        Set<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findByLectureIdAndAttachmentType(lectureId, AttachmentType.FILE);
+        if (lectureAttachments.isEmpty()) {
+            return ResponseEntity.badRequest().build();
+        }
+        if (!validateTemporaryAccessToken(temporaryAccessToken, "merge-pdf")) {
+            // NOTE: this is a special case, because we like to show this error message directly in the browser (without the angular client being active)
+            String errorMessage = "You don't have the access rights for this file! Please login to Artemis and download the attachment in the corresponding attachmentUnit";
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(errorMessage.getBytes());
+        }
```

Comment:
I don’t know if validating an access token is an expensive operation. If not, I think it makes more sense to first validate that the user is allowed to access any resources and only then query the attachments.
```suggestion
        if (!validateTemporaryAccessToken(temporaryAccessToken, "merge-pdf")) {
            // NOTE: this is a special case, because we like to show this error message directly in the browser (without the angular client being active)
            String errorMessage = "You don't have the access rights for this file! Please login to Artemis and download the attachment in the corresponding attachmentUnit";
            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(errorMessage.getBytes());
        }

        Set<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findByLectureIdAndAttachmentType(lectureId, AttachmentType.FILE);
        if (lectureAttachments.isEmpty()) {
            return ResponseEntity.badRequest().build();
        }
```

Reply:
Also, you can change `findByLectureIdAndAttachmentType` with `findByLectureIdAndAttachmentTypeElseThrow`. Inside of this function you can handle the empty case, it can increase the readability.